src/MatrixTypes_Native.f90

```diff
@@ -770,6 +1698,46 @@ SUBROUTINE transpose_TriDiagMatrixType(matrix)
 !> @brief tranpose the matrix
 !> @param matrix declare the matrix type to act on
 !>
+!>
+    SUBROUTINE transpose_DistributedBandedMatrixType(matrix)
+      CHARACTER(LEN=*),PARAMETER :: myName='transpose_DistributedBandedMatrixType'
+      CLASS(DistributedBandedMatrixType),INTENT(INOUT) :: matrix
+      INTEGER(SIK) :: mTmp,i
+
+      CALL eMatrixType%raiseFatalError(modName//'::'//myName// &
+        ' - routine is not implemented!')
+
+    ENDSUBROUTINE transpose_DistributedBandedMatrixType
+!
+!-------------------------------------------------------------------------------
+!> @brief tranpose the matrix
+!> @param matrix declare the matrix type to act on
+!>
+!>
+    SUBROUTINE transpose_BandedMatrixType(matrix)
+      CHARACTER(LEN=*),PARAMETER :: myName='transpose_BandedMatrixType'
+      CLASS(BandedMatrixType),INTENT(INOUT) :: matrix
+      INTEGER(SIK) :: mTmp,i
+
+      REQUIRE(matrix%isInit)
+      REQUIRE(matrix%isAssembled)
+
+      mTmp = matrix%n
+      matrix%n = matrix%m
+      matrix%m = mTmp
```

Comment:
Use the new swap routines?

Reply:
They aren't currently public. I could mark them as public though if you think it's worth it.

Reply:
If you can use it in enough places that makes it worth being public, sure.  Reduces redundant coding.

Reply:
I don't think there are many places that it could be used, and unfortunately it can't be used here because those routines swap 2 elements of an array by index.
So I think it's best to leave it.